Simplenote/Classes/SPAuthViewController.swift

```diff
@@ -696,131 +799,27 @@ extension SPAuthViewController: SPTextInputViewDelegate {
             state.username = textInput.text ?? ""
         case passwordInputView:
             state.password = textInput.text ?? ""
+        case codeInputView:
+            state.code = textInput.text ?? ""
         default:
             break
         }
     }
 
     func textInputShouldReturn(_ textInput: SPTextInputView) -> Bool {
-        switch textInput {
-        case emailInputView:
-            switch performUsernameValidation() {
-            case .success:
-                if mode.isPasswordHidden {
-                    performPrimaryActionIfPossible()
-                } else {
-                    passwordInputView.becomeFirstResponder()
-                }
-
-            case let error:
-                displayEmailValidationWarning(error.description)
-            }
-
-        case passwordInputView:
-            switch performPasswordValidation() {
-            case .success:
-                performPrimaryActionIfPossible()
-
-            case let error:
-                displayPasswordValidationWarning(error.description)
-            }
-
-        default:
-            break
-        }
-
+        performPrimaryActionIfPossible()
```

Comment:
`performPrimaryActionIfPossible` is now responsible for rendering Warnings, when / if needed.

**Plus** we'll never show Username + Password fields, at once, so we're simplifying this